ckan_meta_tester/ckan_meta_tester.py

```diff
@@ -179,8 +179,8 @@ def netkans(self) -> Iterable[Path]:
         return (f for f in Path().rglob('*')
                 if f.is_file() and f.suffix.lower() == '.netkan')
 
-    def branch_commits(self, repo: Repo) -> Iterable[Commit]:
-        return repo.iter_commits(f'{self.get_start_ref()}..{repo.head}')
+    def branch_diff(self, repo: Repo) -> DiffIndex:
+        return repo.head.commit.diff(self.get_start_ref())
```

Comment:
I think we might need to switch the order of the diff, so it's `master.diff(branch)` instead of `branch.diff(master)`. Right now when I try this logic with the branch for KSP-CKAN/NetKAN#8275, it seems to say I'm deleting the files I'm creating:

```
>>> [print(d) for d in r.head.commit.diff('2f3f4464')]
NetKAN/Buffalo-PlayMode-CRP.netkan
=======================================================
lhs: 100644 | 204377c1a2a7370e85433b41084713fd1eb5d375
rhs: None
file deleted in rhs
NetKAN/Buffalo-PlayMode-ClassicStock.netkan
=======================================================
lhs: 100644 | dea7e9e563ffe99ea5906a51faba5eb43541d52e
rhs: None
file deleted in rhs
NetKAN/Buffalo.netkan
=======================================================
lhs: 100644 | 69b54972f51a52ed272c41430d646d4de8c241d2
rhs: 100644 | 234deb1596dd1411b2bb2efb27b11998c4766d1c
```

Reply:
Output of swapped order, for comparison:

```
>>> [print(d) for d in r.commit('2f3f4464').diff(r.head.commit)]
NetKAN/Buffalo-PlayMode-CRP.netkan
=======================================================
lhs: None
rhs: 100644 | 204377c1a2a7370e85433b41084713fd1eb5d375
file added in rhs
NetKAN/Buffalo-PlayMode-ClassicStock.netkan
=======================================================
lhs: None
rhs: 100644 | dea7e9e563ffe99ea5906a51faba5eb43541d52e
file added in rhs
```

Reply:
Confirmed, I think; the created files aren't returned by `paths_from_diff`:

```
>>> [print(x) for x in cmt.paths_from_diff(cmt.branch_diff(r))]
NetKAN/ClassicStockResources.netkan
NetKAN/HooliganLabsAirships.netkan
NetKAN/KiwiTechTree.netkan
NetKAN/WildBlueTools.netkan
NetKAN/Pathfinder.netkan
NetKAN/Heisenberg.netkan
NetKAN/MOLE.netkan
NetKAN/DSEV.netkan
NetKAN/TestLite.netkan
NetKAN/Buffalo.netkan
```

Reply:
Same test after suggested change:

```
>>> [print(x) for x in cmt.paths_from_diff(cmt.branch_diff(r))]
NetKAN/MOLE.netkan
NetKAN/DSEV.netkan
NetKAN/Pathfinder-PlayMode-CRP.netkan
NetKAN/DSEV-PlayMode-ClassicStock.netkan
NetKAN/TechTreeKompacted.netkan
NetKAN/Heisenberg-PlayMode-Simplified.netkan
NetKAN/TestLite.netkan
NetKAN/Buffalo.netkan
NetKAN/WildBlue-PlayMode-CRP.netkan
NetKAN/MOLE-PlayMode-CRP.netkan
NetKAN/Heisenberg-PlayMode-Pristine.netkan
NetKAN/ClassicStockResources.netkan
NetKAN/Pathfinder-PlayMode-Pristine.netkan
NetKAN/MOLE-PlayMode-Pristine.netkan
NetKAN/WildBlue-PlayMode-USI.netkan
NetKAN/WildBlue-PlayMode-Pristine.netkan
NetKAN/KiwiTechTree.netkan
NetKAN/Buffalo-PlayMode-ClassicStock.netkan
NetKAN/HooliganLabsAirships.netkan
NetKAN/WildBlue-PlayMode-Simplified.netkan
NetKAN/WildBlue-PlayMode-ClassicStock.netkan
NetKAN/Pathfinder-PlayMode-ClassicStock.netkan
NetKAN/Buffalo-PlayMode-CRP.netkan
NetKAN/MOLE-PlayMode-ClassicStock.netkan
NetKAN/WildBlueTools.netkan
NetKAN/Pathfinder.netkan
NetKAN/Pathfinder-PlayMode-Simplified.netkan
NetKAN/DSEV-PlayMode-CRP.netkan
NetKAN/Heisenberg-PlayMode-ClassicStock.netkan
NetKAN/Heisenberg.netkan
NetKAN/Heisenberg-PlayMode-CRP.netkan
NetKAN/DSEV-PlayMode-Pristine.netkan
```

Reply:
Ouch, you're right, thanks for catching that.

Reply:
Don't feel bad, I wasn't able to find the commit, but I had the exact same problem earlier with the old code

Reply:
After spending some time looking at them, I came to the conclusion that their docs are at least misleading if not outright wrong on this:
https://gitpython.readthedocs.io/en/stable/tutorial.html#obtaining-diff-information

It contains the following code example:
```
hcommit = repo.head.commit
# Traverse added Diff objects only
for diff_added in hcommit.diff('HEAD~1').iter_change_type('A'):
    print(diff_added)
```

Which is basically `newer.diff(older)` i.e. `branch.diff(master)` whereas you only get the diff the right way round if you do `master.diff(branch)` as you wrote above.

Reply:
Yeah, that is quite misleading. Their documentation format is kind of confusing, I'm not sure where that should be fixed. Here, I guess?

https://github.com/gitpython-developers/GitPython/blob/696e4edd6c6d20d13e53a93759e63c675532af05/test/test_docs.py#L409-L425